tests/acceptance/book-appointment-test.js

```diff
@@ -0,0 +1,445 @@
+        responseText:["2018-06-14", "2018-06-15", "2018-06-16", "2018-06-19", "2018-06-20", "2018-06-21"]}),
+      mockFindAll("gogovan_transport").returns({json: {gogovan_transports: [gogo_van.toJSON({includeId: true})]}}),
+      mockFindAll('order').returns({ json: {orders: [order.toJSON({includeId: true})],
+        packages: [pkg.toJSON({includeId: true})], orders_packages: [ordersPackage.toJSON({includeId: true})]}}),
+      mockFindAll("booking_type").returns({json: {booking_types: [bookingType.toJSON({includeId: true})]}})
+    );
+    requestPurposeUrl = '/request_purpose';
+    clientInfoUrl = `/order/${order.id}/client_information`;
+    goodsDetailsUrl = `/order/${order.id}/goods_details`;
+    appointmentPageUrl = `/order/${order.id}/appointment_details`;
+    confirmBookingUrl = `/order/${order.id}/confirm_booking`;
+    bookingSuccessUrl = `/order/${order.id}/booking_success`;
+  },
+
+  afterEach: function() {
+     // Clear our ajax mocks
+    $.mockjaxSettings.matchInRegistrationOrder = true;
+    mocks.forEach($.mockjax.clear);
+
+    // Stop the app
+    Ember.run(App, 'destroy');
+  }
+});
+
+// home page login flow with book appointment test
+test("should redirect to login page if user not logged in", function(assert) {
+  window.localStorage.removeItem('authToken');
+  assert.expect(2);
+  visit('/');
+  andThen(function() {
+    click(".book-appointment");
+  });
+  andThen(function(){
+    assert.equal(currentURL(), '/login?bookAppointment=true');
+    assert.equal(Ember.$('.title').text().trim(), "Log in/Register");
+  });
+});
+
+// home page account details flow with book appointment test
+test("should redirect to Account details page if user is logged in and account details are incomplete", function(assert) {
+  assert.expect(2);
+  visit('/');
+  andThen(function() {
+    click(".book-appointment");
+  });
+  andThen(function(){
+    assert.equal(currentURL(), "/account_details?bookAppointment=true");
+    assert.equal(Ember.$('.title').text().trim(), "Account Details");
+  });
+});
+
+//Request Purpose Page tests
+test("request purpose page on completely filled should redirect to Goods details if For organisation's own programs is selected", function(assert){
+  orderPurpose1 = {id: 1, purpose_id: 1, order_id: order.id, designation_id: 1};
+  assert.expect(2);
+  visit(requestPurposeUrl);
+  andThen(function(){
+    click(".for-organisation");
+  });
+  andThen(function(){
+    click("#people-count");
+  });
+  andThen(function(){
+    fillIn("#people-count", "2");
+  });
+  andThen(function(){
+    fillIn("#description", "test test test");
+  });
+  andThen(function(){
+    mocks.push(
+      $.mockjax({url: '/api/v1/order*', type: 'POST', status: 201,responseText: {
+        order: order.toJSON({includeId: true}),
+        orders_purposes: [orderPurpose1],
+        user: user.toJSON({includeId: true})}
+      })
+    );
+    click('.button.expand');
```

Comment:
Don't we have id for button ? 🤔 These classes looks common.. 😅 If we can add some id to that button then that will be good I think If I have missed it.

Reply:
there's no id, on the button, but since there's only one button on the page, it works fine, i'll add id to each page.

Reply:
Okay.. Sure.. That will be great. 👍